src/lib/webhook.ts

```diff
@@ -135,36 +141,29 @@ export class WebhookController extends EventEmitter {
   private async startSubscriptions(): Promise<void> {
     if (this.options.client && this.options.url && this.options.channel &&
       (this.options.messageEventHandler || this.options.messageStatusEventHandler)) {
-      const subscriptions = await this.options.client.listSubscriptions();
-
-      let messageSubscriptionAlreadySet = false;
-      let messageStatusSubscriptionAlreadySet = false;
-
-      if (subscriptions) {
-        subscriptions.map((subscription: ISubscription) => {
-          if (subscription.status === 'ACTIVE' && subscription.webhook.url === this.options.url && subscription.criteria.channel === this.options.channel) {
-            if (subscription.eventType === 'MESSAGE') {
-              messageSubscriptionAlreadySet = true;
-            } else if (subscription.eventType === 'MESSAGE_STATUS') {
-              messageStatusSubscriptionAlreadySet = true;
-            }
-          }
-        });
-      }
 
-      if (this.options.messageEventHandler && !messageSubscriptionAlreadySet) {
+      if (this.options.messageEventHandler) {
         const subscription = new MessageSubscription(
           {
             url: this.options.url,
           },
           {
             channel: this.options.channel,
+            direction: this.options.direction,
           },
         );
 
-        await this.options.client.createSubscription(subscription);
+        try {
+          await this.options.client.createSubscription(subscription);
+        } catch (error) {
+          if (error.httpStatusCode === HTTP_CONFLICT_ERROR) {
+            this.logger.debug('Message subscription already exists.');
+          } else {
+            throw error;
+          }
+        }
       }
-      if (this.options.messageStatusEventHandler && !messageStatusSubscriptionAlreadySet) {
+      if (this.options.messageStatusEventHandler) {
```

Comment:
Is it not necessary to inform the direction property on line 172? Following the idea from line 152

Reply:
Maybe is better to unify this similar blocks and avoid duplications